core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt

```diff
@@ -165,4 +170,273 @@ object GraphQLUtils {
+                                    history.add(ob)//add the union objects to the history
+                                    addEdge(element.tableFieldType, ob, unionTag, graph) // from the union type object to its objects
+                                    if (history.count { it == ob } <= 1) {
+                                        constructGraph(state, ob, "", graph, history, objectFieldsHistory)
+                                        history.remove(ob)
+                                    } else {
+                                        history.remove(ob)
+                                    }
+                                }
+                                history.remove(element.tableFieldType)// remove the union type object to the history: the normal case (not cycle)
+                            } else {
+                                history.remove(element.tableFieldType)// remove the union type object to the history:the cycle case
+                            }
+                        } else {
+                            if (element.kindOfTableFieldType.toString().toLowerCase() == "interface") {
+                                history.add(element.tableFieldType)// add interface type object to the history
+                                addEdge(entryPoint, element.tableFieldType, element.tableField, graph)//from a node to the interface type object
+                                if (history.count { it == element.tableFieldType } <= 1) {// the interface type object already treated
+                                    constructGraph(state, element.tableFieldType, "", graph, history, objectFieldsHistory)//construct the base fields of the interface
+                                    element.interfaceTypes.forEach { ob ->
+                                        history.add(ob)//add the interface objects to the history
+                                        addEdge(element.tableFieldType, ob, interfaceTag, graph) // from the interface type object to its objects
+                                        if (history.count { it == ob } <= 1) {
+                                            constructGraph(state, ob, "", graph, history, objectFieldsHistory)
+                                            history.remove(ob)
+                                        } else {
+                                            history.remove(ob)
+                                        }
+                                    }
+                                    history.remove(element.tableFieldType)// remove the interface type object to the history:the normal case (not cycle)
+                                } else {
+                                    history.remove(element.tableFieldType)// remove the interface type object to the history:the cycle case
+                                }
+                            }
+                        }
+                    }
+
+                }
+            }
+        }
+        return graph
```

Comment:
at least 7 levels of nesting... code can be simplified... :) eg, splitting in private methods and/or better handling of if/else

Reply:
-Already done.